oomph-plugin-notes.php

```diff
@@ -0,0 +1,194 @@
+Plugin URI: http://www.oomphinc.com/plugins-modules/oomph-plugin-notes
+Description: Add usage notes to your plugins
+Author: Ben Doherty @ Oomph, Inc.
+Version: 0.1.0
+Author URI: http://www.oomphinc.com/people/bdoherty/
+License: GPLv2 or later
+
+		Copyright © 2016 Oomph, Inc. <http://oomphinc.com>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/**
+ * @package Oomph Plugin Notes
+ */
+class Oomph_Plugin_Notes {
+	// Store plugin notes in this option
+	const OPTION_NAME = 'plugin_notes';
+
+	// Capability to see hidden tags
+	const CAPABILITY = 'activate_plugins';
+
+	// Nonce action
+	const NONCE_ACTION = 'oomph-plugin-notes';
+
+	// Filters and actions used by this class
+	var $filters = array(
+		'plugin_row_meta' => array( 20, 'add_plugin_notes' ),
+		'wp_ajax_oomph-plugin-notes-save' => 'save_plugin_notes',
+		'admin_enqueue_scripts' => 'enqueue_scripts',
+		'admin_head' => 'add_styles',
+	);
+
+	/**
+	 * Register filters and assets
+	 */
+	function __construct() {
+		$this->filter_map( 'add_filter' );
+	}
+
+	/**
+	 * Unregister all filters on object destruction
+	 */
+	function __destruct() {
+		$this->filter_map( 'remove_filter' );
+	}
+
+	/**
+	 * Enqueue scripts used in this module on the plugin screen and add
+	 * script data.
+	 *
+	 * @action admin_enqueue_scripts
+	 */
+	function enqueue_scripts() {
+		global $current_screen;
+
+		if( $current_screen->base !== 'plugins' ) {
+			return;
+		}
+
+		wp_enqueue_script( 'oomph-plugin-notes', plugins_url( 'oomph-plugin-notes.js', __FILE__ ), array(), 1, true );
+		wp_localize_script( 'oomph-plugin-notes', 'OPN', array(
+			'nonce' => wp_create_nonce( self::NONCE_ACTION ),
+			'text' => array(
+				'save' => __( 'Save' ),
+				'cancel' => __( 'Cancel' )
+			)
+		) );
+	}
+
+	/**
+	 * Add styles used in this module to the header
+	 *
+	 * @action admin_head
+	 */
+	function add_styles() {
+		global $current_screen;
+
+		if( $current_screen->base !== 'plugins' ) {
+			return;
+		}
+
+		echo <<<STYLES
+<style type="text/css">
+.plugin-notes-edit { padding-left: 1em; white-space: nowrap; color: #999; cursor: pointer; }
+</style>
+STYLES;
+
+	}
+
+	/**
+	 * Return a link that can be clicked to edit the current notes
+	 */
+	function edit_link() {
+		if( current_user_can( self::CAPABILITY ) ) {
+			return '<span class="plugin-notes-edit">' . __( 'edit notes' ) . '</span>';
```

Comment:
The `$domain` parameter defaults to `'default'`, so technically, yes, this particular string should have a domain specific to this plugin, but 'Save' and 'Cancel' are pretty generic. It's stubbing it out for a later version :)